### PR TITLE
Reorganize tutorial section in the documentation sidebar

### LINKED ISSUE #1191: Redesign the tutorial section on the documentation sidebar
We already have a long list of tutorials in the "User Guide" part. I expect more tutorials will be added in the future, and the length of the sidebar will increase. I think we need to think about how to reorganize the learning material and how to shorten the sidebar.

image

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR implements the suggestions proposed in #1191  using a folding structure with beginner, intermediate, and advanced sections for the various tutorials.

Here is the appearance with one of the sections expanded:

image

Fixes #1191 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]
Looks good to me. Makes the whole structure much more tidy!

**Comment by willschlitzer** [comment]
This looks like a good idea (and reminds me that I need to motivate to work on #770 :grimacing: )

**Comment by seisman** [comment]
I agree with putting "Gallery", "Projections", "Tutorials" and "External Resources" in the "USER GUIDE" part, but I don't like the subfolder structure for the tutorials. For any new tutorials, we may waste time debating if a tutorial belongs to the "beginner", "intermediate", or "advanced" groups. Personally, I prefer to have all tutorials in the same folder.

**Comment by weiji14** [comment]
> I agree with putting "Gallery", "Projections", "Tutorials" and "External Resources" in the "USER GUIDE" part, but I don't like the subfolder structure for the tutorials. For any new tutorials, we may waste time debating if a tutorial belongs to the "beginner", "intermediate", or "advanced" groups. Personally, I prefer to have all tutorials in the same folder.

Maybe just have 2 sections (Basics and Advanced as suggested by @core-man in https://github.com/GenericMappingTools/pygmt/issues/1191#issuecomment-817129541)? I do think there's a bit of subjectivity on what counts as beginner/intermediate/advanced, so better to make things more simple.

**Comment by meghanrjones** [comment]
I consolidated intermediate and advanced tutorials and added to the contributing guide that new tutorials should go in `examples/tutorials/advanced`, since we have the beginner topics pretty well covered.

**Comment by willschlitzer** [comment]
Should we put "Tutorials" at the top of the "User Guide" section? That would be my pick; I think it should be at the top of the section.

**Review by seisman — APPROVED** [review]
Looks good to me.

**Review by willschlitzer — APPROVED** [review]

File: examples/tutorials/advanced/date_time_charts.py [hunk 1]
```diff
@@ -285,7 +285,7 @@
 # Starting off with ``WS``, adding this string means that only
 # Western/Left (**W**) and Southern/Bottom (**S**) borders of
 # the plot will be shown. For more information on this, please
-# refer to :doc:`frame instructions </tutorials/frames>`.
+# refer to :doc:`frame instructions </tutorials/beginner/frames>`.
 #
 # The other important item in the ``frame`` list is
 # ``"sxa1Of1D"``. This string modifies the secondary
```
Comment (weiji14) [inline examples/tutorials/advanced/date_time_charts.py:291]: Not really related, but could you update L14 of this `date_time_charts.py` file to get a proper thumbnail? Right now it's showing up like this:

image

File: examples/tutorials/README.txt [hunk 2]
```diff
@@ -1,2 +1,6 @@
+.. _tutorials:
+
 Tutorials
 =========
+
+These tutorials teach how to complete various tasks using PyGMT.
```
Comment (weiji14) [inline examples/tutorials/README.txt:6]:
```suggestion
These examples teach us how to complete various tasks using PyGMT!
```

File: examples/tutorials/beginner/README.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Beginner Tutorials
+------------------
```
Comment (weiji14) [inline examples/tutorials/beginner/README.txt:2]: Could this "Beginner Tutorials" heading be shortened to be just "Beginner"? Same with "Advanced Tutorials" -> "Advanced". Right now it looks like this:

image

Also, just wondering if "Basics" would be a better term than "Beginner" (similar to what @core-man used at https://github.com/GenericMappingTools/pygmt/issues/1191#issuecomment-817129541), considering that PyGMT experts still need to use these basic building blocks sometimes!

**Review by weiji14 — COMMENTED** [review]
Thanks @meghanrjones! Just some comments.

**Comment by seisman** [comment]
Can we merge this PR now?

**Review by weiji14 — APPROVED** [review]
Yes please, go for it!

**Pull request merged**